lab-beeksma/server.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+const uuidv4 = require('uuid/v4');
+const net = require('net');
+const PORT = process.env.PORT || 3000;
+const server = net.createServer();
+const EE = require('events');
+const ee = new EE();
+
+const clientPool = [];
+
+function Client(socket){
+  this.nickname = 'default';
+  this.id = uuidv4();
+  this.socket = socket;
+}
+
+ee.on('@all', function (sender, message) {
+  if(sender.nickname === 'default'){
```

Comment:
What if I want `default` as my nickname? Could just as easily use `null` or `undefined` for this scenario.

I wonder if it should really be `@all` and `@dm`'s responsibility to check for this. Consider how you might refactor to let these event handlers focus on handling pre-validated commands.